src/masternodes/mn_rpc.cpp

```diff
@@ -1558,32 +2005,675 @@ UniValue accounttoutxos(const JSONRPCRequest& request) {
+                        {
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                   },
+
+               },
+               RPCResult{
+                       "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+               },
+               RPCExamples{
+                       HelpExampleCli("updatetoken", "\"{\"pool\":\"POOL\",\"status\":true,"
+                                                     "\"commission\":0.01,\"ownerAddress\":\"Address\"}\" "
+                                                     "\"[{\"txid\":\"id\",\"vout\":0}]\"")
+                       + HelpExampleRpc("updatetoken", "\"{\"pool\":\"POOL\",\"status\":true,"
+                                                       "\"commission\":0.01,\"ownerAddress\":\"Address\"}\" "
+                                                       "\"[{\"txid\":\"id\",\"vout\":0}]\"")
+               },
+    }.Check(request);
+
+    if (pwallet->chain().isInitialBlockDownload()) {
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD,
+                           "Cannot create transactions while still in Initial Block Download");
+    }
+    pwallet->BlockUntilSyncedToCurrentChain();
+
+    RPCTypeCheck(request.params, {UniValue::VOBJ, UniValue::VARR}, true);
+
+    bool status = true;
+    CAmount commission = -1;
+    CScript ownerAddress;
+    UniValue metaObj = request.params[0].get_obj();
+
+    std::string const poolStr = trim_ws(metaObj["pool"].getValStr());
+    CTxDestination ownerDest;
+    DCT_ID poolId;
+    {
+        LOCK(cs_main);
+        auto token = pcustomcsview->GetTokenGuessId(poolStr, poolId);
+        if (!token) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Pool %s does not exist!", poolStr));
+        }
+
+        auto pool = pcustomcsview->GetPoolPair(poolId);
+        if (!pool) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Pool %s does not exist!", poolStr));
+        }
+        status = pool->status;
+    }
+
+    if (!metaObj["status"].isNull()) {
+        status = metaObj["status"].getBool();
+    }
+    if (!metaObj["commission"].isNull()) {
+        commission = AmountFromValue(metaObj["commission"]);
+    }
+    if (!metaObj["ownerAddress"].isNull()) {
+        ownerAddress = DecodeScript(metaObj["ownerAddress"].getValStr());
+    }
+
+    CMutableTransaction rawTx;
+
+    for(std::set<CScript>::iterator it = Params().GetConsensus().foundationMembers.begin(); it != Params().GetConsensus().foundationMembers.end() && rawTx.vin.size() == 0; it++)
```

Comment:
Why not use for (auto& it : Params().GetConsensus().foundationMembers)

Reply:
the same as previous